src/screens/MainApp/DiagnoseRequest/ImageReview/index.js

```diff
@@ -0,0 +1,136 @@
+import React, { useState, useRef, useEffect } from 'react'
+import { View, FlatList } from 'react-native'
+import Carousel, { Pagination } from 'react-native-snap-carousel'
+import Icon from 'react-native-vector-icons/FontAwesome5'
+import { Button } from '~/components'
+import NavigationService from '~/navigationService'
+import ConfirmButton from './ConfirmButton'
+import CarouselImage from './CarouselImage'
+import FlatListImage from './FlatListImage'
+import { requestImages, newImages } from '~/constants/mocks'
+import { MIN_IMAGES, MAX_IMAGES, ADD_MORE_IMAGES_URI } from './constants'
+import styles, { ITEM_HEIGHT, ITEM_WIDTH, CAROUSEL_SLIDER_WIDTH } from './styles'
+
+const ImageVisualization = (props) => {
+  const [activeIndex, setActiveIndex] = useState(0)
+  const [submitError, setSubmitError] = useState(false)
+  const [images, setImages] = useState(requestImages.slice())
+  const carouselRef = useRef()
+  const flatListRef = useRef()
+
+  const setFlatListItem = (index) => {
+    setActiveIndex(index)
+    flatListRef.current.scrollToIndex({ index: index })
+  }
+
+  const setCarouselItem = (index) => {
+    setActiveIndex(index)
+    carouselRef.current.snapToItem(index)
+  }
+
+  const confirmRequest = () => {}
+
+  const getMorePhotos = () => {
+    const filterDuplicates = newImages.filter(element => !images.includes(element))
+    setImages([...images, filterDuplicates[0]])
+  }
+
+  const FlatListItem = ({ item }) => {
+    const itemIndex = images.indexOf(item)
+    const lastIndex = images.length - 1
+    const selected = images[activeIndex] === item
+    if (itemIndex === lastIndex && images.length < MAX_IMAGES) {
+      return (
+        <>
+          <FlatListImage selected={selected} item={item} onPress={() => setCarouselItem(images.indexOf(item))} />
+          <FlatListImage item={ADD_MORE_IMAGES_URI} onPress={getMorePhotos} />
+        </>
+      )
+    }
+    return (
+      <FlatListImage selected={selected} item={item} onPress={() => setCarouselItem(images.indexOf(item))} />
+    )
+  }
+
+  useEffect(() => {
+    if (images.length < MIN_IMAGES) {
+      setSubmitError(true)
+    } else {
+      setSubmitError(false)
+    }
+
+    const deleteImage = () => {
```

Comment:
I think this function could be more readable if you extract the content of the conditions out to separate functions. If I read correctly (I did it quickly) the first condition deletes a single image and the second one multiple images, am I right? If so, it could be much clearer this way:

```javascript
if (images.length === 1) {
  deleteSingleImage(images)
} else {
  deleteMultipleImages(images)
}
```

Related but to exactly for this case, just material to think about: https://www.youtube.com/watch?v=os6ssw1qylU

Reply:
Both paths delete a single image. The division was necessary to Navigate to another screen when no images are present (the user deleted all of them) and because 

`images.filter((val, index) => index !== activeIndex)` 

would fail when `images.length === 1` as the current selected index is the only one.

I had problems implementing this though, is there a convention when modifying an array that is part of the state?

I wanted to use `images.splice(activeIndex, 1)` but it was just modifying the state directly. I also thought about doing a copy and then splice on the copy, but it seemed overkill.

Regardless of this, I think I can extract functions like that.